service/history/queues/dlq_writer.go

```diff
@@ -99,5 +109,17 @@ func (q *DLQWriter) WriteTaskToDLQ(ctx context.Context, sourceCluster, targetClu
 	if err != nil {
 		return fmt.Errorf("%w: %v", ErrSendTaskToDLQ, err)
 	}
+	q.metricsHandler.Counter(metrics.DLQWrites.GetMetricName()).Record(1)
+	ns, err := q.namespaceRegistry.GetNamespaceByID(namespace.ID(task.GetNamespaceID()))
+	if err != nil {
+		q.logger.Warn("Failed to get namespace name", tag.WorkflowNamespace(task.GetNamespaceID()))
```

Comment:
Couple of things:
1. I think the error message should state that this happened while trying to write the task to the DLQ.
2. Please include the error itself in the warning message.
3. We should still log that the task was enqueued to the DLQ when this happens, but we should use the namespace ID tag instead of the namespace name.
4. We should add a test case for this when the registry fails.

Reply:
Makes sense. I have made the changes.